wip-http-post.md

```diff
@@ -0,0 +1,138 @@
+<pre>
+  WIP: WIP-XXXX
+  Layer: Consensus (hard fork)
+  Title: Support HTTP-POST in RADON
+  Authors: Tomasz Polaczyk <tomasz@witnet.foundation>
+  Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+  Status: Draft
+  Type: Standards Track
+  Created: 2021-12-02
+  License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+RADON supports HTTP retrieval sources, but only using the HTTP verb GET. Some APIs require the use of the HTTP verb POST. This proposal specifies support for HTTP POST requests.
+
+## Motivation and rationale
+
+Same as above?
+
+## Specification
+
+TL;DR: new fields in protobuf schema:
+
+```diff
+ enum RADType {
+     Unknown = 0;
+     HttpGet = 1;
+     Rng = 2;
++    HttpPost = 3;
+ }
+ message RADRetrieve {
+     RADType kind = 1;
+     string url = 2;
+     // TODO: RADScript should maybe be a type?
+     bytes script = 3;
++    // Body of HTTP-POST request
++    bytes body = 4;
++    // Extra headers for HTTP-GET and HTTP-POST requests
++    repeated StringPair headers = 5;
+ }
```

Comment:
Oh man how hard this sucks. Really regretting not having a different `RadRretrieve` for each `RadType` :cry:

Reply:
I don't like the direction this is is taking. But I guess if we change the shape of a data request, we'll have to deal with migrations :bomb:

Reply:
You mean it would have been better to have something like this?

```rust
enum RadRetrieve {
    Rng { script },
    HttpGet { url, script, headers },
    HttpPost { url, script, body, headers },
}
```

Maybe we can do some conversion and use that enum in Rust code, but yeah the protobuf schema is hard to change. Also keep in mind that in protobuf all fields are optional, so adding more fields doesn't increase the message size. But it's true that it will make messages harder to validate.